networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_cfg_syncer.py

```diff
@@ -0,0 +1,207 @@
+            conn = self.driver._get_connection()
+
+        LOG.info(_LI("*************************"))
+        LOG.info(_LI("neutron router db records"))
+        for router_id, router in six.iteritems(router_id_dict):
+            LOG.info(_LI("ROUTER_ID: %s"), router_id)
+
+        LOG.info(_LI("\n"))
+
+        for segment_id, intf_list in six.iteritems(intf_segment_dict):
+            LOG.info(_LI("SEGMENT_ID: %s"), (segment_id))
+            for intf in intf_list:
+                dev_owner = intf['device_owner']
+                dev_id = intf['device_id'][0:6]
+                ip_addr = intf['fixed_ips'][0]['ip_address']
+                intf_info = "    INTF: %s, %s, %s" % \
+                            (ip_addr, dev_id, dev_owner)
+                LOG.info(intf_info)
+
+        running_cfg = self.get_running_config(conn)
+        parsed_cfg = HTParser(running_cfg)
+
+        invalid_cfg = []
+
+        invalid_cfg += self.clean_snat(conn,
+                                       router_id_dict,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg)
+
+        invalid_cfg += self.clean_nat_pool_overload(conn,
+                                                    router_id_dict,
+                                                    intf_segment_dict,
+                                                    segment_nat_dict,
+                                                    parsed_cfg)
+
+        invalid_cfg += self.clean_nat_pool(conn,
+                                           router_id_dict,
+                                           intf_segment_dict,
+                                           segment_nat_dict,
+                                           parsed_cfg)
+
+        if (cfg.CONF.multi_region.enable_multi_region):
+            set_route_regex = SET_ROUTE_MULTI_REGION_REGEX
+        else:
+            set_route_regex = SET_ROUTE_REGEX
+
+        invalid_cfg += self.clean_routes(conn,
+                                       router_id_dict,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg,
+                                       set_route_regex)
+        invalid_cfg += self.clean_acls(conn,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg)
+
+        invalid_cfg += self.clean_interfaces(conn,
+                                             intf_segment_dict,
+                                             segment_nat_dict,
+                                             parsed_cfg)
+
+        invalid_cfg += self.clean_vrfs(conn, router_id_dict, parsed_cfg)
+        LOG.debug("invalid_cfg = %s " % (pp.pformat(invalid_cfg)))
+        return invalid_cfg
+
+    def clean_routes(self,
```

Comment:
this function is almost the same as clean_default_route() of the parent class....Can we just re-use the one in the parent's class then?